PSPrintSDK/OLProduct.m

```diff
@@ -20,58 +20,28 @@ @implementation OLProduct
 
 +(NSArray *)products{
     static NSMutableArray *products = nil;
-    if (!products) {
-        NSArray *templates = [OLProductTemplate templates];
-        products = [[NSMutableArray alloc] initWithCapacity:[templates count]];
-        for (OLProductTemplate *template in templates){
-            if (template.enabled){
-                [products addObject:[[OLProduct alloc] initWithTemplate:template]];
-            }
+    NSArray *templates = [OLProductTemplate templates];
+    products = [[NSMutableArray alloc] initWithCapacity:[templates count]];
+    for (OLProductTemplate *template in templates){
+        if (template.enabled){
+            [products addObject:[[OLProduct alloc] initWithTemplate:template]];
         }
     }
     
     return products;
 }
 
--(CGSize)serverImageSize{
-    CGFloat pointsToPixels = 1.38;
-    if (self.templateType == kOLTemplateTypeLargeFormatA1){
-            return CGSizeMake(1564.724409288 * pointsToPixels, 2264.881889531 * pointsToPixels);
-    }
-    else if (self.templateType == kOLTemplateTypeLargeFormatA2){
-        return CGSizeMake(1088.503936896 * pointsToPixels, 1581.732283302 * pointsToPixels);
-    }
-    else if (self.templateType == kOLTemplateTypeLargeFormatA3){
-        return CGSizeMake(785.196850313 * pointsToPixels, 1133.8582676 * pointsToPixels);
-    }
-    else if (self.templateType == kOLTemplateTypePolaroids){
-            return CGSizeMake(945, 945);
-    }
-    else if (self.templateType == kOLTemplateTypeSquares){
-        return CGSizeMake(1111, 1111);
-    }
-    else if (self.templateType == kOLTemplateTypeMagnets){
-        return CGSizeMake(733, 733);
-    }
-    else if (self.templateType == kOLTemplateTypeMiniSquares){
-        return CGSizeMake(750, 750);
-    }
-    else if (self.templateType == kOLTemplateTypeStickersSquare){
-        return CGSizeMake(702, 702);
-    }
-    else if (self.templateType == kOLTemplateTypeStickersCircle){
-        return CGSizeMake(748, 748);
-    }
-    else if (self.templateType == kOLTemplateTypeMiniPolaroids){
-        return CGSizeMake(656, 656);
-    }
-    else if (self.templateType == kOLTemplateTypeFrame || self.templateType == kOLTemplateTypeFrame2x2 || self.templateType == kOLTemplateTypeFrame3x3 || self.templateType == kOLTemplateTypeFrame4x4){
-        return CGSizeMake(1651, 1651);
-    }
-    else{
-        return CGSizeMake(0, 0);
-    }
+- (UIColor *)labelColor{
+    return self.productTemplate.labelColor;
+}
+
+#warning TODO
```

Comment:
Yup, currently it is not returned from the server and that might take some time as we don't have the pixel dimensions for everything.
There won't be any problems if we merge now.
It will be used for 1) asset size warnings and 2) resizing the image to the exact resolution before sending to server. Right now we send the image cropped to the correct aspect ratio but not resized.